tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -581,6 +581,10 @@ bool DeclResultIdMapper::createStageVars(const DeclaratorDecl *decl,
     if (varId == 0)
       return false;
 
+    if (sigPoint->GetSignatureKind() ==
```

Comment:
Wait, is this correct? I'm not able to find the description for `Patch`. Based on the reading of `Patch` in SPIR-V spec, it seems to me that it should be decorated on the type `InputPatch` or `OutputPatch` is parmeterized on? We are doing it on all flattened variables of patch constant interface here.

Reply:
Discussed offline. It seems that ``Patch`` decoration must be applied to patch-constant data (outputs of the patch constant function) which are outputs of hull and inputs of domain shaders.